### DISCUSSION TITLE
Input at the start of onboarding?

### DISCUSSION THREAD

**Discussion by LRSFC-DanJ** [IBM/mac-ibm-notifications#73] [Q&A] [answered]
I've been looking for an alternative to our current deployment splash screen tool DEPnotify and someone suggested this project.

I've been trying to get my head round how things work but it's a little confusing with ongoing communication taking place over named pipes, and none of the examples match my use case particularly closely.

What we do currently is prompt for a computer name (text field) and a machine role (dropdown with a choice of staff/student/server, but radio buttons would also be acceptable). The information entered in these choices is then used to set the computer name and the value of an extension attribute "Machine Role" in Jamf.

We then need to be able to display Jamf log messages about package deployment for our policies that deploy direct from our Jamf installation, and then Munki log messages for our policies that deploy software using DataJAR Auto-Update (which uses Munki under the hood).

Is this possible? How would we do it?

**Comment by SMartorelli** [comment, marked as answer]
Hi @LRSFC-DanJ, if I have correctly understood your workflow I would suggest you to use a pop-up to ask for the first two information you need to start the process (use `input` and `dropdown` accessory views, it is possible to show two different accessory views with one pop-up) and then to use two different pop-ups with progress bar accessory view (or one if you want to fit the Jamf and Munki log messages in a single process) to show to the user what is going on in background. 
Please refer to the [wiki](https://github.com/IBM/mac-ibm-notifications/wiki) to understand how to properly format the [payload for the accessory views](https://github.com/IBM/mac-ibm-notifications/wiki/Pop-up-UI-Accessory-Views) and how to dynamically feed the progress bar.
We also have some [script examples](https://github.com/IBM/mac-ibm-notifications/wiki/Script-Examples) for progress bar.

**Reply by LRSFC-DanJ** [reply]
Thanks for your reply. The part I don't quite understand is how I get the values entered/chosen in the input and dropdown out of the notifier app so that I can use them in my script?

Also, it sounds like you are saying that the accessory views are not available in onboarding mode, so I'd need to use the popup mode initially? Is it possible to change modes while the notifier app is running, or would I need to kill the process and restart it with the new mode options?

**Reply by SMartorelli** [reply]
IBM Notifier simply print in the stdout of the process the value of the accessory views when the user end the process (Click on the main button) so you can easily do:
`value=$(/Applications/IBM\ Notifier/Contents/MacOS/IBM\ Notifier -type popup -title "This is a title" -subtitle "This is a subtitle" -main_button_label "Send" -accessory_view_type input -accessory_view_payload "/placeholder This is a placeholder")`
If you have 2 accessory views the two different value are separated by the "new line" character.

Accessory Views are not currently available in the Onboarding UI as you can see from the [related documentation](https://github.com/IBM/mac-ibm-notifications/wiki/Onboarding-UI-Workflow). We're working on integrate those but no ETA yet.
